libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -733,8 +732,19 @@ impl NodeAPI for Greenlight {
             .invoices
             .first()
             .cloned()
-            .and_then(|res| res.bolt11);
-        Ok(found_bolt11)
+            .and_then(|invoice| {
+                invoice.bolt11.map(|bolt11| {
+                    serde_json::from_str::<InvoiceLabel>(&invoice.label)
```

Comment:
If the label doesn't exist (or json parse fail) this function will return None? Do we want to return the invoice with payer amount equals to the invoice amount in that case?

Reply:
Good catch!/We should return the invoice with payer amount equals None in that case.

Reply:
Done in https://github.com/breez/breez-sdk/pull/862/commits/fd3fa20f91e396d47c3694ac7187725f1fe5004d